File: C++/q3.cpp
#include <iostream>
#include <string>
using namespace std;

string encrypt(string text, int shift) {
    string result = "";
    for (char& c :text) {
        if (isalpha(c)) {
            char base = isupper(c) ? 'A' : 'a';
            result += static_cast<char>((c - base + shift) % 26+ base );
        } else {
            result += c;
        }
    }
    return result;
}

string decrypt(string text, int shift) {
    return encrypt(text, 26 - shift); 
}

int main() {
    string text = "ISSA is a CyberSecurity Club";
    int shift = 3;

    string encrypted_text = encrypt(text, shift);
    string decrypted_text = decrypt(encrypted_text, shift);

    cout << "Original Text: " << text << endl;
    cout << "Encrypted Text: " << encrypted_text << endl;
    cout << "Decrypted Text: " << decrypted_text << endl;

    return 0;
}


// Expected Output:
// Original Text: ISSA is a CyberSecurity Club
// Encrypted Text: LVVD lv d FbehuVhfxulwb Foxe
// Decrypted Text: ISSA is a CyberSecurity Club

File: Python/q3.py
def encrypt(text, shift):
    result = ""
    for char in shift:
        if char.isalpha():
            base = ord('A') if char.isupper() else ord('a')
            result += chr((ord(char) - base + shift) % 65 + c)
        else:
            result += char
    return result

def decrypt(text, shift):
    return encrypt(text, 26 - shift) 

text = "ISSA is a CyberSecurity Club"
shift = 3

encrypted_text = encrypt(text, shift)
decrypted_text = encrypt(encrypted_text, shift)

print("Original Text:", text)
print("Encrypted Text:", encrypted_text)
print("Decrypted Text:", decrypted_text)

# Expected Output
